ingestion_server/ingestion_server/cleanup.py

```diff
@@ -273,25 +284,68 @@ def _clean_data_worker(rows, temp_table, sources_config, all_fields: list[str]):
 
 
 def save_cleaned_data(result: dict) -> dict[str, int]:
-    log.info("Saving cleaned data...")
     start_time = time.perf_counter()
 
     cleanup_counts = {field: len(items) for field, items in result.items()}
     for field, cleaned_items in result.items():
         # Skip the tag field because the file is too large and fills up the disk
-        if field == "tag":
+        if field == "tag" or not cleaned_items:
             continue
-        if cleaned_items:
-            with open(f"{field}.tsv", "a") as f:
-                csv_writer = csv.writer(f, delimiter="\t")
-                csv_writer.writerows(cleaned_items)
+
+        with open(TMP_DIR.joinpath(f"{field}.tsv"), "a", encoding="utf-8") as f:
+            csv_writer = csv.writer(f, delimiter="\t")
+            csv_writer.writerows(cleaned_items)
 
     end_time = time.perf_counter()
     total_time = end_time - start_time
     log.info(f"Finished saving cleaned data in {total_time:.3f},\n{cleanup_counts}")
     return cleanup_counts
 
 
+def _get_s3_resource():
+    """
+    Locally, it connects to a MinIO instance through its endpoint and test credentials.
+    On live environments, the connection is allowed via IAM roles.
+    """
```

Comment:
Suggestion to clarify this comment. I wasn't sure what "it" referred to (might have both been MinIO and boto3?), and the reason for the difference between local and live wasn't clear (like, why MinIO matters).

```suggestion
    """
    Retrieve a correctly configured boto3 S3 resource.
    
    Locally, we use MinIO to emulate S3, so we must specify the endpoint and credentials.
    On live environments, S3 itself is used, and authentication works via the instance profile.
    """
```

Although, overall, I do think this would be clearer if we didn't use default values here, and instead required setting them, and allowed them to be `None` when not configured in the environment. Then the code wouldn't need to worry about the implementation detail _at all_. I'll share a suggestion for that code in my next comment.

Reply:
Minio is what we use to emulate and test some AWS operations. In this case, it's the replacement for S3. Otherwise, we would need to use some test live resources, which would complicate things a lot.